packages/desktop/src/renderer/index.tsx

```diff
@@ -102,6 +104,10 @@ store.subscribe(() => {
         {}
     );
     if (JSON.stringify(appState) !== JSON.stringify(oldAppState)) {
+        if (appState.RECENT_ANNOTATIONS[0] == undefined) {
```

Comment:
This is the part I do not like. For some reason when a new user opens the app appstate.RECENT_ANNOTATIONS defaults to a list with one defined element rather than an empty list. This breaks when trying to persist the state because the typing does not match. While resetting on the first persist does work its hacky and would be better to fix at the source.

Reply:
I still think we should find a fix for this rather than special case it if at all possible. My immediate hunch is this is because of the way you are grabbing the values from the persisted storage on load based on your description.
I believe we discussed this, but did you check to see if this:
```
    const recentAnnotations =
        persistedConfig && persistedConfig[PersistedConfigKeys.RecentAnnotations];
```
Could be causing an array like so `[undefined]`? Probably worth trying to see if:
```
    const recentAnnotations = persistedConfig?.[PersistedConfigKeys.RecentAnnotations]?.length
           ? persistedConfig?.[PersistedConfigKeys.RecentAnnotations]
           : [];
```
fixes it?